app/javascript/articles/components/SaveButton.jsx

```diff
@@ -8,20 +8,23 @@ export class SaveButton extends Component {
 
     const { isBookmarked } = props;
 
-    this.state = { buttonText: isBookmarked ? 'Saved' : 'Save' };
+    this.state = {
+      buttonText: isBookmarked ? 'Saved' : 'Save',
+      hovering: false,
+    };
   }
 
   render() {
-    const { buttonText } = this.state;
+    //const { hovering } = this.state.hovering;
 
     const { article, isBookmarked, onClick } = this.props;
 
     const mouseOver = (_e) => {
-      this.setState({ buttonText: isBookmarked ? 'Unsave' : 'Save' });
+      this.setState({ hovering: true });
     };
 
     const mouseOut = (_e) => {
-      this.setState({ buttonText: isBookmarked ? 'Saved' : 'Save' });
+      this.setState({ hovering: false });
     };
 
     const handleClick = (_e) => {
```

Comment:
A bug was here because I wasn't doing setting the `isBookmarked` component state property

```js
      this.setState({
        buttonText: isBookmarked ? 'Save' : 'Saved',
        isBookmarked: !isBookmarked,
      });
```

Reply:
Hey i put your suggestions in the code and I'm getting the same problem we initially had
```
 const handleClick = (_e) => {
      onClick(_e);
      this.setState({ 
        buttonText: isBookmarked ? 'Save' : 'Saved',
        isBookmarked: !isBookmarked
       });
       
    };
```

Reply:
I also tried putting `isBookmarked` outside of `this.setState` by doing 
```
this.setState({ 
        buttonText: isBookmarked ? 'Save' : 'Saved',
       });
isBookmarked = !isBookmarked;
```

Reply:
I got rid of the mouseOver event and replaced it with the mouseMove event too